collector-kinesis/src/main/java/zipkin/collector/kinesis/KinesisSpanProcessor.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.kinesis;
+
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessor;
+import com.amazonaws.services.kinesis.clientlibrary.types.InitializationInput;
+import com.amazonaws.services.kinesis.clientlibrary.types.ProcessRecordsInput;
+import com.amazonaws.services.kinesis.clientlibrary.types.ShutdownInput;
+import com.amazonaws.services.kinesis.model.Record;
+import zipkin.Codec;
+import zipkin.collector.Collector;
+import zipkin.internal.Nullable;
+import zipkin.storage.Callback;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class KinesisSpanProcessor implements IRecordProcessor {
+
+    private final Collector collector;
+
+    KinesisSpanProcessor(Collector collector) {
+        this.collector = collector;
+    }
+
+    @Override
+    public void initialize(InitializationInput initializationInput) {
+    }
+
+    @Override
+    public void processRecords(ProcessRecordsInput processRecordsInput) {
+        List<byte[]> spans = processRecordsInput.getRecords().stream()
```

Comment:
Yeah, I'll back it down to Java 7. I went with the zipkin server version forgetting about those pesky java7ers